core/src/main/java/com/linecorp/armeria/common/ContextAwareBiConsumer.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common;
+
+import java.util.function.BiConsumer;
+
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+/**
+ * A delegating {@link BiConsumer} that makes sure an underlying BiConsumer is
+ * executed within the {@link RequestContext}.
+ */
+@UnstableApi
+public interface ContextAwareBiConsumer<T, U> extends BiConsumer<T, U>, ContextHolder {
+
+    /**
+     * Returns a new {@link ContextAwareBiConsumer} that sets the specified {@link RequestContext}
+     * before executing an underlying {@link BiConsumer}.
+     */
+    static <T, U> ContextAwareBiConsumer of(RequestContext context, BiConsumer<T, U> action) {
```

Comment:
This adds two public APIs for doing the same thing I think

```
ContextAwareBiConsumer.of(ctx, action)
ctx.makeContextAware(action)
```

Probably it's better to avoid these factory methods since the context has them already

Reply:
I was mostly following the pattern that's already in place with things like [ContextAwareExecutor](https://github.com/line/armeria/blob/main/core/src/main/java/com/linecorp/armeria/common/ContextAwareExecutor.java#L33).

Don't feel strongly about it either way but since we can't (yet) change the type returned from `ctx.makeX`, maybe leaving another API (`ContextAwareBiConsumer`) around isn't a bad idea to help users avoid type casting.